File: commandAndMessage/main/BasicGui.java
package commandAndMessage.main;

import javax.swing.*;

import java.awt.*;
import java.awt.event.KeyListener;
import java.util.ArrayList;
import java.util.List;

public class BasicGui extends JFrame {
    private List<String> commands;
    private final JLabel label;
    private final JLabel label2;

    public BasicGui() {
            commands = new ArrayList<>();
            setupBasicLayout();
            label= new JLabel();
            label2= new JLabel();

            label.setText("MyText");
            label2.setText("MyText");

            add(label);
            add(label2);

            label2.setForeground(this.getBackground());

            label.grabFocus();
    }

    @Override
    public void setVisible(boolean b) {
        super.setVisible(b);
        label.grabFocus();
    }

    public final void addListener(KeyListener keyListener) {
        this.addKeyListener(keyListener);
        label.addKeyListener(keyListener);
        label2.addKeyListener(keyListener);
        label.grabFocus();
    }

    public final void setText(String text) {
        label.setText(text);
        label2.setText(text);
        if (label.hasFocus()) {
            label2.grabFocus();
        } else if (label2.hasFocus()) {
            label.grabFocus();
        } else {
            label.grabFocus();
        }
    }

    private final void setupBasicLayout() {
        setTitle("Enceladus");
        setSize(600, 400);
        setLocationRelativeTo(null);
        setDefaultCloseOperation(EXIT_ON_CLOSE);
        setLayout(new GridLayout(0,1));
    }
}

File: commandAndMessage/main/BasicGui.kt
package commandAndMessage.main

import javax.swing.*

import java.awt.*
import java.awt.event.KeyListener
import java.util.ArrayList

class BasicGui : JFrame() {
    private val commands: List<String>
    private val label: JLabel
    private val label2: JLabel

    init {
        commands = ArrayList()
        setupBasicLayout()
        label = JLabel()
        label2 = JLabel()

        label.text = "MyText"
        label2.text = "MyText"

        add(label)
        add(label2)

        label2.foreground = this.background

        label.grabFocus()
    }

    override fun setVisible(b: Boolean) {
        super.setVisible(b)
        label.grabFocus()
    }

    fun addListener(keyListener: KeyListener) {
        this.addKeyListener(keyListener)
        label.addKeyListener(keyListener)
        label2.addKeyListener(keyListener)
        label.grabFocus()
    }

    fun setText(text: String) {
        label.text = text
        label2.text = text
        if (label.hasFocus()) {
            label2.grabFocus()
        } else if (label2.hasFocus()) {
            label.grabFocus()
        } else {
            label.grabFocus()
        }
    }

    private fun setupBasicLayout() {
        title = "Enceladus"
        setSize(600, 400)
        setLocationRelativeTo(null)
        defaultCloseOperation = JFrame.EXIT_ON_CLOSE
        layout = GridLayout(0, 1)
    }
}